File: gcc/testsuite/gfortran.dg/gomp/loop-transforms/unroll-non-rect-1.f90
subroutine test
  implicit none

  integer :: i, j, k
  !$omp target parallel do collapse(2) ! { dg-error {invalid OpenMP non-rectangular loop step; '\(2 - 1\) \* 1' is not a multiple of loop 2 step '5'} }
  do i = -300, 100
    !$omp unroll partial
    do j = i,i*2
      call dummy (i)
    end do
   end do

  !$omp target parallel do collapse(3) ! { dg-error {invalid OpenMP non-rectangular loop step; '\(2 - 1\) \* 1' is not a multiple of loop 3 step '5'} }
  do i = -300, 100
    do j = 1,10
       !$omp unroll partial
       do k = j,j*2 + 1
      call dummy (i)
    end do
   end do
  end do

  !$omp unroll full
  do i = -3, 5
    do j = 1,10
       do k = j,j*2 + 1
      call dummy (i)
    end do
   end do
  end do
end subroutine

File: gcc/testsuite/c-c++-common/gomp/loop-transforms/unroll-non-rect-1.c
extern void dummy (int);

void
test1 ()
{
#pragma omp target parallel for collapse(2)
  for (int i = -300; i != 100; ++i)
#pragma omp unroll partial(2)
    for (int j = i * 2; j <= i * 4 + 1; ++j)
      dummy (i);

#pragma omp target parallel for collapse(3)
  for (int i = -300; i != 100; ++i)
    for (int j = i; j != i * 2; ++j)
    #pragma omp unroll partial
    for (int k = 2; k != 100; ++k)
      dummy (i);

#pragma omp unroll full
  for (int i = -300; i != 100; ++i)
    for (int j = i; j != i * 2; ++j)
    for (int k = 2; k != 100; ++k)
      dummy (i);

  for (int i = -300; i != 100; ++i)
#pragma omp unroll full
    for (int j = i; j != i + 10; ++j)
    for (int k = 2; k != 100; ++k)
      dummy (i);

  for (int i = -300; i != 100; ++i)
#pragma omp unroll full
    for (int j = i; j != i + 10; ++j)
    for (int k = j; k != 100; ++k)
      dummy (i);
}

